test/integration_test.py

```diff
@@ -1035,3 +1043,78 @@ def test_local_filtered_index_endpoints(self):
                                              catalog=self.catalog)).url
         response = requests.get(url)
         self.assertEqual(200, response.status_code)
+
+
+class CanBundleScriptIntegrationTest(AzulTestCase):
+    scripts_root = f'{config.project_root}/scripts'
+
+    def _test_catalog(self, catalog: config.Catalog):
+        with self.subTest(catalog=catalog.name,
+                          repository=catalog.plugins['repository']):
+            fqid = self.bundle_fqid(catalog.name)
+            with tempfile.TemporaryDirectory() as d:
+                self._can_bundle(source=str(fqid.source.spec),
+                                 uuid=fqid.uuid,
+                                 version=fqid.version,
+                                 output_dir=d)
+                parts = ['manifest', 'metadata']
+                expected_files = sorted(f'{fqid.uuid}.{part}.json' for part in parts)
+                generated_files = sorted(os.listdir(d))
+                self.assertListEqual(generated_files, expected_files)
+
+    def test_can_bundle_configured_catalogs(self):
+        for catalog_name, catalog in config.catalogs.items():
+            if catalog.is_integration_test_catalog:
+                self._test_catalog(catalog)
+                break
```

Comment:
I'm confused by the break. With the break you only test the first IT catalog. But then why is the function's name plural? And why, within `_test_catalog`, do you use subTest? I suppose subtest adds helpful context to the test, but a log statement would do that too.

Also, the way this is written, the test will pass if no IT catalog is configured. Is that the behavior we want? Alternatively you could do a comprehension on config.catalogs with the same condition and select the desired item with `first`, which will raise if the iterable is empty.

Reply:
> I'm confused by the break.

Rightfully so, I added this to speed up debugging and never meant for it to stick around :sweat_smile:

Reply:
> why, within _test_catalog, do you use subTest?

`subTest` makes it a lot easier to tell at a glance which catalogs succeed or fail when running the tests via the Pycharm UI. I like to use it whenever the test can be broken down to pieces that succeed or fail independently of each other.

Reply:
> Also, the way this is written, the test will pass if no IT catalog is configured. Is that the behavior we want?

I can't find any precedent in the `integration_test` module for failing when no IT catalogs are configured.